cluster-autoscaler/metrics/metrics.go

```diff
@@ -304,6 +304,14 @@ var (
 		},
 	)
 
+	podsPerOwnerRefOverflowCount = k8smetrics.NewCounter(
+		&k8smetrics.CounterOpts{
+			Namespace: caNamespace,
+			Name:      "pods_per_owner_ref_overflow_count",
+			Help:      "Number of pods that were rejected from internal caching, due to high number of different pods under the same owner reference.",
```

Comment:
I find this definition of metric a bit surprising:
 * If we're after pod count wouldn't a gauge be a better choice? With counter it's inherently a function of both pod count and how often we run the logic (so CA loop time).
 * I guess the first 10 pods weren't technically 'rejected from internal caching', but the fact that we're reporting the number of affected pods - 10 feels arbitrary.
 * Aren't we double-counting the pods in any loop where we trigger scale-up?

Maybe having a gauge counting controllers with enough pods to disable caching would be easier to count precisely and provide a more meaningful value? Admittedly, it may not matter very much, since the main use of this seems to be checking if the number is >0 or not.

Reply:
Good point on scraping interval, I haven't thought about that. Since we count controllers in filterOutSchedulable, the value should not go down for the duration of node startup which I think should be long enough for scraping? I guess you agree, since I see you changed it to a gauge? You've much more experience with metrics than I do, so I'll defer to your judgement.